src/detect-engine-mpm.c

```diff
@@ -736,6 +736,10 @@ static void PopulateMpmHelperAddPattern(MpmCtx *mpm_ctx,
         }
     }
 
+    if (cd->flags & (DETECT_CONTENT_DEPTH_BE | DETECT_CONTENT_OFFSET_BE)) {
```

Comment:
can you add a comment here on the 'why'?

Also: style nits:
1. indent looks off
2. commit msg does not have colon after detect/mpm.

Reply:
Will fix.